deeplearning4j-scaleout/deeplearning4j-nlp/src/main/java/org/deeplearning4j/models/glove/GloveWeightLookupTable.java

```diff
@@ -69,21 +69,21 @@ public void resetWeights(boolean reset) {
             this.rng = Nd4j.getRandom();
 
         //note the +2 which is the unk vocab word and the bias
-        if(syn0 == null || syn0 != null && reset) {
+        if(syn0 == null || reset) {
             syn0 = Nd4j.rand(new int[]{vocab.numWords() + 1, vectorLength}, rng).subi(0.5).divi((double) vectorLength);
```

Comment:
@nyghtowl again `syn0 != null` check is redundant, because if it was `null`, then the first part of the expression would be `true` and it wouldn't evaluate the second part, so it would always be `true` and again that is redundant and less readable/confusing.